src/gui/ControllerView.cpp

```diff
@@ -72,27 +71,17 @@ ControllerView::ControllerView( Controller * _model, QWidget * _parent ) :
 
 	hBox->addWidget(controlsButton);
 
-	m_nameLabel = new QLabel(_model->name(), this);
+	m_nameLabel = new QLabel(model->name(), this);
 	vBoxLayout->addWidget(m_nameLabel);
 
 
-	m_controllerDlg = getController()->createDialog( getGUI()->mainWindow()->workspace() );
-
-	m_subWindow = getGUI()->mainWindow()->addWindowedWidget( m_controllerDlg );
-	
-	Qt::WindowFlags flags = m_subWindow->windowFlags();
-	flags &= ~Qt::WindowMaximizeButtonHint;
-	m_subWindow->setWindowFlags( flags );
-	m_subWindow->setFixedSize( m_subWindow->size() );
-
-	m_subWindow->setWindowIcon( m_controllerDlg->windowIcon() );
-
-	connect( m_controllerDlg, SIGNAL(closed()),
-		this, SLOT(closeControls()));
+	m_controllerDlg = getController()->createDialog(getGUI()->mainWindow()->workspace());
 
+	m_subWindow = getGUI()->mainWindow()->addWindowedWidget(m_controllerDlg);
```

Comment:
`addWindowedWidget()` takes ownership of `m_controllerDlg`. We could ditch `m_subWindow` and call `m_controllerDlg->parentWidget()->show()` directly.